tsdb/meta.go

```diff
@@ -1810,15 +1830,26 @@ func MarshalTags(tags map[string]string) []byte {
 // TagKeys returns a list of the measurement's tag names.
 func (m *Measurement) TagKeys() []string {
 	m.mu.RLock()
-	defer m.mu.RUnlock()
 	keys := make([]string, 0, len(m.seriesByTagKeyValue))
 	for k := range m.seriesByTagKeyValue {
 		keys = append(keys, k)
 	}
+	m.mu.RUnlock()
 	sort.Strings(keys)
 	return keys
 }
 
+func (m *Measurement) TagKeysBytes() [][]byte {
```

Comment:
I don't think we should be copying/allocating every tag key once per minute in the mean time, though.